sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala

```diff
@@ -174,7 +174,7 @@ class ResolveCatalogs(val catalogManager: CatalogManager)
         c.tableSchema,
         // convert the bucket spec and add it as a transform
         c.partitioning ++ c.bucketSpec.map(_.asTransform),
-        convertTableProperties(c.properties, c.options, c.location, c.comment, c.provider),
+        convertTableProperties(c.properties, c.options, c.location, c.comment, Some(c.provider)),
```

Comment:
Can the parser return this as `null`? If so, shouldn't these be `Option(c.provider)` so that they don't return null providers

Reply:
Seems that's required https://github.com/apache/spark/pull/27650/files#diff-9847f5cef7cf7fbc5830fbc6b779ee10L2803 . So should be fine.

Reply:
https://github.com/cloud-fan/spark/blob/6685dc5dbbbea0f4cd069f1a5e92ccb70033f493/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala#L2826

This is how the parser gets the provider. I don't think it can be null.